sparkmagic/sparkmagic/livyclientlib/command.py

```diff
@@ -42,6 +42,19 @@ def execute(self, session):
             response = session.http_client.post_statement(session.id, data)
             statement_id = response[u'id']
             output = self._get_statement_output(session, statement_id)
+        except KeyboardInterrupt as e:
+            msg =  u'Interrupted by user'
+            if statement_id >= 0:
+                response = session.http_client.cancel_statement(session.id, statement_id)
+                if u'msg' not in response or response[u'msg'] != u'canceled':
```

Comment:
`livy` always returns `{ "msg": "canceled" }` if it is successful. If it is not successful, `session.http_client.cancel_statement` will raise an `HttpClientException`, so I don't see a reason to check the response here. On success, we should re-raise a helpful error message to inform the user